src/viam/rpc/dial.py

```diff
@@ -313,12 +313,22 @@ async def _dial_direct(address: str, options: Optional[DialOptions] = None) -> C
     if insecure:
         ctx = None
     else:
-        ctx = ssl.create_default_context(purpose=ssl.Purpose.SERVER_AUTH)
+        is_local_host = host is not None and (host.startswith("localhost") or host.startswith("0.0.0.0") or host.startswith("127."))
```

Comment:
Not positive about how prioritization of bool operators works in python, added the parens around the `or` statements for explicitness but I'm not sure if it's actually necessary?

Reply:
I never am either. I like this because it's explicit